### ISSUE TITLE
Issue with relative paths on filesystem boundary with mounted directories

### ISSUE THREAD

**Issue by jlsalmon** [cirruslabs/tart#554]
I realise that this is probably an issue with Virtualization.Framework or virtiofs and likely nothing to do with Tart itself, but I thought I'd post this here in case anyone has any ideas.

We've noticed some strange behaviour when using relative paths that hit the filesystem boundary of a mounted directory. For example, running `tart run --dir test:/some/host/dir` and trying to compile a very simple C++ program with relative header includes:

```
admin@admins-Virtual-Machine test % pwd
/Volumes/My Shard Files/test
```

```
admin@admins-Virtual-Machine test % cat main.cpp
#include "h1.h"
#include "dir/h2.h"

int main() { return 0; }
```
```
admin@admins-Virtual-Machine test % cat h1.h
#pragma once

int a;
```
```
admin@admins-Virtual-Machine test % cat dir/h2.h
#pragma once

#include "../h1.h"

int b;
```

```
admin@admins-Virtual-Machine test % clang main.cpp
In file included from main.cpp:2:
In file included from ./dir/h2.h:3:
./dir/../h1.h:3:5: error: redefinition of 'a'
int a;
    ^
./h1.h:3:5: note: previous definition is here
int a;
    ^
1 error generated.
```

However, if we move all the files into another directory which is one level deeper, it works:

```
admin@admins-Virtual-Machine test % mkdir nested
admin@admins-Virtual-Machine test % mv main.cpp h1.h dir/ nested/
admin@admins-Virtual-Machine test % cd nested
admin@admins-Virtual-Machine nested % clang main.cpp     <-- this works
```

**Comment by fkorotkov** [comment]
Which version of macOS you have on the host and guest? We are getting reports that prior to macOS Ventura 13.3 the virtiofs integration flaking with different issue but seems on 13.3+ it was improved a lot.

**Comment by fkorotkov** [comment]
Fixed by #555. Please try mounting a single directory without a name.
